controllers/common.go

```diff
@@ -775,17 +774,31 @@ func (r *MultiClusterHubReconciler) ensureNoClusterManagementAddOn(m *operatorv1
 	}
 
 	err = r.Client.Delete(context.TODO(), clusterMgmtAddon)
-	if err != nil && !errors.IsNotFound(err) {
+	if errors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	}
+
+	if err != nil {
 		r.Log.Error(err, fmt.Sprintf("Error deleting ClusterManagementAddOn CR"))
+
 		return ctrl.Result{Requeue: true}, err
```

Comment:
Requeue: true is ignored when an error is returned. Let's just not set it since returning an error uses the exponential backoff in controller-runtime.
```suggestion
		return ctrl.Result{}, err
```